code/modules/pda/cart_apps.dm

```diff
@@ -375,47 +376,35 @@
 			var/direction = get_dir(pda, M)
 			MopData[++MopData.len] = list ("x" = ml.x, "y" = ml.y, "dir" = uppertext(dir2text(direction)), "status" = M.reagents.total_volume ? "Wet" : "Dry")
 
-	if(!MopData.len)
-		MopData[++MopData.len] = list("x" = 0, "y" = 0, dir=null, status = null)
-
-
 	var/BucketData[0]
 	for(var/obj/structure/mopbucket/B in GLOB.janitorial_equipment)
 		var/turf/bl = get_turf(B)
 		if(bl)
 			if(bl.z != cl.z)
 				continue
 			var/direction = get_dir(pda,B)
-			BucketData[++BucketData.len] = list ("x" = bl.x, "y" = bl.y, "dir" = uppertext(dir2text(direction)), "status" = B.reagents.total_volume/100)
-
-	if(!BucketData.len)
-		BucketData[++BucketData.len] = list("x" = 0, "y" = 0, dir=null, status = null)
+			BucketData[++BucketData.len] = list ("x" = bl.x, "y" = bl.y, "dir" = uppertext(dir2text(direction)), "volume" = B.reagents.total_volume, "max_volume" = B.reagents.maximum_volume)
 
 	var/CbotData[0]
-	for(var/mob/living/simple_animal/bot/cleanbot/B in GLOB.simple_animals)
+	for(var/mob/living/simple_animal/bot/cleanbot/B in GLOB.simple_animals[AI_ON])
```

Comment:
iterating over the bot list would be even better.

Reply:
Good to know, thanks